libraries/cli/lib/util/Configuration.ts

```diff
@@ -0,0 +1,132 @@
+
+export type GeneralOptions = {
+  config: string;
+  modules: string[];
+  userInterface: string;
+};
+
+export type StorageOptions = {
+  syntestDirectory: string;
+  tempSyntestDirectory: string;
+};
+
+export type LoggingOptions = {
+  logDirectory: string;
+  consoleLogLevel: string;
+  logToFile: string[];
+};
+
+export type BaseOptions = GeneralOptions & StorageOptions & LoggingOptions;
+
+export enum OptionGroups {
+  General = "General Options:",
+  Storage = "Storage Options:",
+  Logging = "Logging Options:",
+}
+
+export class Configuration {
+  static configureUsage() {
+    return (
+      Yargs.usage(`Usage: syntest <tool> <command> [options]`)
+        // TODO examples
+        .epilog("visit https://syntest.org for more documentation")
+    );
+  }
+
+  static configureOptions(yargs: Yargs.Argv) {
+    return (
+      yargs
+        .option("config", {
+          alias: ["c"],
+          default: ".syntest.json",
+          description: "The syntest configuration file",
+          group: OptionGroups.General,
+          hidden: false,
+          config: true,
+          type: "string",
+        })
+        .option("modules", {
+          alias: ["m"],
+          array: true,
+          default: [],
+          description: "List of dependencies or paths to modules to load",
+          group: OptionGroups.General,
+          hidden: false,
+          type: "string",
+        })
+        // ui
+        .option("user-interface", {
+          alias: [],
+          default: "regular",
+          description: "The user interface you use",
+          group: OptionGroups.General,
+          hidden: false,
+          type: "string",
+        })
+        // storage
+        .options("syntest-directory", {
+          alias: [],
+          default: "syntest",
+          description: "The path where everything should be saved",
+          group: OptionGroups.Storage,
+          hidden: false,
+          normalize: true,
+          type: "string",
+        })
+        .options("temp-syntest-directory", {
+          alias: [],
+          default: ".syntest",
+          description: "The path where all temporary files should be saved",
+          group: OptionGroups.Storage,
+          hidden: false,
+          normalize: true,
+          type: "string",
+        })
+        // logging options
+        .options("log-directory", {
+          alias: [],
+          default: "logs",
+          description: "The path where the logs should be saved",
+          group: OptionGroups.Logging,
+          hidden: false,
+          normalize: true,
+          type: "string",
+        })
+        .options("console-log-level", {
+          alias: [],
+          choices: ["debug", "error", "warn", "info", "verbose", "silly"],
+          default: "debug",
+          description: "Log level of the tool",
+          group: OptionGroups.Logging,
+          hidden: false,
+          type: "string",
+        })
+        .options("log-to-file", {
```

Comment:
Maybe we should call this: `file-log-level`? log-to-file seems more like a boolean to me

Reply:
Haha, agree I fixed this in #238 already.